rskj-core/src/main/java/co/rsk/trie/TrieStoreImpl.java

```diff
@@ -31,24 +35,22 @@
  */
 public class TrieStoreImpl implements TrieStore {
 
-    // a key value data source to use
     private KeyValueDataSource store;
 
+    /** Weak references are removed once the tries are garbage collected */
+    private Set<Trie> savedTries = Collections.newSetFromMap(new WeakHashMap<>());
```

Comment:
You deserve this
![dca](https://user-images.githubusercontent.com/79943/61641117-edf5bf00-ac74-11e9-99db-9a9485fea076.png)

Reply:
😍